File: backend/request-service/src/main/java/com/hlrms/requestservice/dto/RequestResponseDto.java
package com.hlrms.requestservice.dto;

import com.hlrms.requestservice.entity.RequestStatus;

import java.time.Instant;
import java.util.UUID;

public record RequestResponseDto(
    UUID id,
    String idempotencyKey,
    String requestType,
    String payload,
    RequestStatus status,
    String result,
    String errorMessage,
    Instant createdAt,
    Instant updatedAt,
    Instant completedAt,
    Long version
) {
}

File: android-app/app/src/main/java/com/hlrms/mobile/data/remote/request/RequestResponseDto.kt
package com.hlrms.mobile.data.remote.request

data class RequestResponseDto(
    val id: String,
    val idempotencyKey: String,
    val requestType: String,
    val payload: String,
    val status: String,
    val result: String?,
    val errorMessage: String?,
    val createdAt: String?,
    val updatedAt: String?,
    val completedAt: String?,
    val version: Long?
)